functions/_pure_prompt_nixdevshell.fish

```diff
@@ -0,0 +1,17 @@
+function _pure_prompt_nixdevshell \
+    --description "Indicate if nix develop shell is activated"
+
+    if set --query pure_enable_nixdevshell;
+        and test "$pure_enable_nixdevshell" = true
+
+        set --local nixdevshell ''
+        set --local nixdevshell_color (_pure_set_color $pure_color_nixdevshell)
+        if test -n "$IN_NIX_SHELL"
+            set nixdevshell $IN_NIX_SHELL
+        end
+
+        if test -n $nixdevshell
+            echo "$pure_symbol_nixdevshell_prefix$nixdevshell_color$nixdevshell"
+        end
+    end
```

Comment:
We can do it in one go:

```suggestion
    if set --query pure_enable_nixdevshell;
        and test "$pure_enable_nixdevshell" = true
        and test -n "$IN_NIX_SHELL"
	        set --local nixdevshell_color (_pure_set_color $pure_color_nixdevshell)

            echo "$pure_symbol_nixdevshell_prefix$nixdevshell_color$IN_NIX_SHELL"
        end
    end
```

Reply:
Thanks for your clean suggestion. Fixed!